zebra-chain/src/transaction/unmined.rs

```diff
@@ -160,7 +164,10 @@ impl From<Transaction> for UnminedTx {
     fn from(transaction: Transaction) -> Self {
         Self {
             id: (&transaction).into(),
-            transaction: Arc::new(transaction),
+            transaction: Arc::new(transaction.clone()),
+            size: transaction
+                .zcash_serialized_size()
+                .expect("all transactions have a size"),
```

Comment:
Here's how we can get the size without cloning the transaction.
(Otherwise we could allocate a lot of memory for a big transaction, then throw it away once we've counted it.)

```suggestion
            size: transaction
                .zcash_serialized_size()
                .expect("all transactions have a size"),
            transaction: Arc::new(transaction),
```